File: MiniProtobufClient/ClientMessageDispatcher.cpp
#include "ClientMessageDispatcher.h"
#include "SigninProcessor.h"
#include "MoveProcessor.h"

using namespace BaseCmd;

ClientMessageDispatcher::ClientMessageDispatcher()
{
	rspSignin rspSignin;
	m_ProcessorMap[rspSignin.GetTypeName()] = std::make_shared<SigninProcessor>();


	rspMove rspMove;
	m_ProcessorMap[rspMove.GetTypeName()] = std::make_shared<MoveProcessor>();
}


ClientMessageDispatcher::~ClientMessageDispatcher()
{
}

File: AndroidProj/app/src/main/java/com/dd/androidprotobuf/ClientMessageDispatcher.java
package com.dd.androidprotobuf;

import com.google.protobuf.GeneratedMessageLite;

import java.util.Dictionary;

import BaseCmd.Cmd;

public class ClientMessageDispatcher extends  BaseMessageDispatcher {


    public  ClientMessageDispatcher()
    {
        Cmd.rspSignin rspSignin = Cmd.rspSignin.newBuilder().build();
        m_ProcessorMap.put(rspSignin.getClass().toString(), new SigninProcessor());


        Cmd.rspMove rspMove = Cmd.rspMove.newBuilder().build();
        m_ProcessorMap.put(rspMove.getClass().toString(),new MoveProcessor());
    }

}
